### PR TITLE
Signup page

### PR DESCRIPTION
**Pull request by arshtsingh**
[signup page #10](https://github.com/beckett-media/comics-pricing-app/issues/10)

### CODE REVIEW

File: backend/src/routes/user.route.ts [hunk 1]
```diff
@@ -14,12 +14,11 @@ userRoutes.post(
...
+  // TODO: verif all this is correct (use joi probably)
   const out = await userService.signup(
     req.body.email,
-    req.body.username,
-    req.body.password
+    req.body.name
   )
   // res.cookie(LOGIN_TOKEN, out.token, { maxAge: out.maxAge })
-  res.send("Logged In")
+  res.send("Signed Up")
```
Comment (randomicons) [inline backend/src/routes/user.route.ts]: send json
res.json

File: backend/src/routes/user.route.ts [hunk 2]
```diff
@@ -14,12 +14,11 @@ userRoutes.post(
 )
 
 userRoutes.post("/signup", async (req: RequestWithBody<any>, res) => {
```
Comment (randomicons) [inline backend/src/routes/user.route.ts]: name it /addToWaitList  and not signup

File: backend/src/services/user.service.ts [hunk 3]
```diff
@@ -1,7 +1,13 @@
 import jwt from "jsonwebtoken"
-import { HttpCode } from "../constants/httpCode"
+import {HttpCode} from "../constants/httpCode"
+import { sql } from "../loader"
 import VError from "verror"
 
+type Emails = {
```
Comment (randomicons) [inline backend/src/services/user.service.ts]: type Email probably

File: backend/src/services/user.service.ts [hunk 4]
```diff
@@ -47,32 +53,23 @@ export const login = async (email: string, password: string) => {
...
-  //   "User already exists"
-  // )
+export const signup = async (email: string, name: string) => {
+  try {
+    return sql<Emails[]>`
+        INSERT INTO emails
+            (email, name)
+        VALUES
+            (${email}, ${name})
+        ON CONFLICT (email) DO NOTHING
```
Comment (randomicons) [inline backend/src/services/user.service.ts:63]: Looks good. I like the extra spaces around lol. I might start doing that

File: frontend/src/pages/Home.tsx [hunk 5]
```diff
@@ -1,10 +1,66 @@
...
+  const [name, setName] = useState("")
+  const [email, setEmail] = useState("")
+
+  const submit = async (event: React.SyntheticEvent) => {
+    event.preventDefault()
+    try {
+      const emailEntry = {
+        email: email,
+        name: name,
+      }
```
Comment (randomicons) [inline frontend/src/pages/Home.tsx]: const emailEntry = {email, name}

File: frontend/src/pages/Home.tsx [hunk 6]
```diff
@@ -15,8 +71,9 @@ export const Home = () => {
   const { data, error } = useSWR("/api/testAPI")
   return (
     <div className="absolute flex h-full w-full items-center justify-center gap-10 bg-slate-800">
-      <Block>{"hello, 😉."}</Block>
+      <Block>{"Get the latest from Comic Pricing App!"}</Block>
```
Comment (randomicons) [inline frontend/src/pages/Home.tsx]: :(

File: frontend/src/pages/Home.tsx [hunk 7]
```diff
@@ -1,10 +1,66 @@
...
+        name: name,
+      }
+      const res = await axios.post("/api/user/signup", emailEntry)
+    } catch (e) {
+      console.log("Failed to add the new emails.")
+    }
+  }
+
+  return (
+    <div className="rounded-2xl shadow-xl hover:shadow-2xl flex h-1/2 w-1/2 flex items-center justify-center">
```
Comment (randomicons) [inline frontend/src/pages/Home.tsx]: styles look good but probably not worth adding at this point

Comment (arshtsingh) [reply frontend/src/pages/Home.tsx]: keep it for now until wireframes :/

**Review by mkxl — APPROVED** [review]

**Pull request merged**